app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java

```diff
@@ -176,15 +138,74 @@ public void onDetach() {
     @Override
     public void onDestroyView() {
         super.onDestroyView();
-        MessageBroadcastReceiver.unregisterReceiver(getActivity(), mUserLocationReceiver);
-        mLocationFetcher.unregisterReceiver(mLocationReceiver);
+        if (mUserLocationReceiver != null) {
```

Comment:
Actually we don't. We spoke about it together and understood its the right place for unregistreation.